### PR TITLE
chore: initial baseline implementation

### PR DESCRIPTION
**Pull request by shjwudp**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/target
+Cargo.lock
```
Comment (NOBLES5E) [inline .gitignore:2]: Cargo.lock should be commited

File: README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+modified: 2021-08-04T13:41:18.756Z
+---
```
Comment (NOBLES5E) [inline README.md:3]: remove

File: README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+## Install
```
Comment (NOBLES5E) [inline README.md:5]: Need complete instructions on how to use this.

File: cc/Makefile [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#
+# Copyright (c) 2015-2019, NVIDIA CORPORATION. All rights reserved.
```
Comment (NOBLES5E) [inline cc/Makefile:2]: fix license

File: cc/bagua_net.cc [hunk 5]
```diff
@@ -0,0 +1,404 @@
+/*************************************************************************
+ * Copyright (c) 2015-2019, NVIDIA CORPORATION. All rights reserved.
```
Comment (NOBLES5E) [inline cc/bagua_net.cc:2]: fix license

File: src/bagua_net.rs [hunk 6]
```diff
@@ -0,0 +1,559 @@
...
+        // println!(
+        //     "isend data.len={}, send_comm_id={}, local={:?}, peer={:?}",
+        //     data.len(),
+        //     send_comm_id,
+        //     (*send_stream).local_addr().unwrap(),
+        //     (*send_stream).peer_addr().unwrap(),
+        // );
+
+        let id = self.socket_request_next_id;
+        // println!(
```
Comment (NOBLES5E) [inline src/bagua_net.rs:377]: debug code should not be committed to master, similar for other places.

File: Cargo.toml [hunk 7]
```diff
@@ -0,0 +1,18 @@
+[package]
+name = "bagua-net"
+version = "0.1.0"
+authors = ["shjwudp"]
+edition = "2018"
```
Comment (NOBLES5E) [inline Cargo.toml:5]: add MIT license, and put your full name in authors (with email address)

**Review by NOBLES5E — CHANGES_REQUESTED** [review]
see comments

File: README.md (hunk 3 above)
Comment (shjwudp) [reply README.md:1]: DONE, add test example

File: Cargo.toml (hunk 7 above)
Comment (shjwudp) [reply Cargo.toml:5]: DONE

**Comment by shjwudp** [comment]
@NOBLES5E The benchmark data has been updated to the README. Can we make the baseline become the master?

**Pull request merged**